main.tf

```diff
@@ -381,7 +381,7 @@ resource "google_compute_security_policy" "policy" {
 
       match {
         expr {
-          expression = rule.value["exclude_ip"] == null ? "evaluateThreatIntelligence('${rule.value["feed"]}')" : "evaluateThreatIntelligence('${rule.value["feed"]}',${rule.value["exclude_ip"]})"
+          expression = lookup(rule.value, "exclude_ip", null) == null ? "evaluateThreatIntelligence('${rule.value["feed"]}')" : "evaluateThreatIntelligence('${rule.value["feed"]}', [%{for index, i in rule.value["exclude_ip"]}'${i}'%{if index < length(rule.value["exclude_ip"]) - 1},%{endif}%{endfor}])" # using string directive
```

Comment:
Have you tried `formatlist` instead of the directive? I'm not completely sure if it'll work but may if you set up a local variable with the `formatlist` functional and just interpolate it here instead of the directive to make it more readable?

Reply:
`formatlist` helps adding string to each list member. Here I need to interpolate whole list. Unlike string interpolation there is no option for list interpolation in terraform. That is why I used string directive to interpolate the whole list. Here is a sample of what I am trying to do:

```
variable "mylist" {
  type = list(strings)
  default = ["abc", "xyz"]
}

output "interpolate_list" {
  value = "my list is ${var.mylist}"
}
```

It fails with following error:

```
Error: Invalid type specification
  on main.tf line 28, in variable "mylist":
  28:   type = list(strings)
The keyword "strings" is not a valid type specification.
```

Reply:
Changed exclude_ip to string instead of a list. Users will have to provide input in format:

```
"['47.100.100.100', '47.189.12.139']"
```